src/python/wntr_vis.py

```diff
@@ -114,24 +123,87 @@ def build_graph(wn, time=1, wnsol=None):
         Gnx.nodes[node]['fillcolor'] = clrstr[1:-1] # remove brackets from string
         if clr[2] < 0.6:
             Gnx.nodes[node]["fontcolor"] = "white"
+    return
+
+
+def link_attrs(wn, Gnx):
+    """
+    Add/change link attributes suitable for visualization
+    """
+    # loop over controls to find pipes with shutoff valves
```

Comment:
Alternatively, we can directly retrieve the shutoff valve names if their valve ids are known - all shutoff valves in the wntr object are named as 'valve'+'#VALVE ID#'.

Reply:
> all shutoff valves in the wntr object are named as 'valve'+'#VALVE ID#'.
I don't think this is true. My test case has been `Richmond_standard.inp` in PowerWaterExamples. The shutoff valve identified by the `CONTROLS` section has the name `v1E`, but that is the name given in the inp file, and `valve` is not prepended to the name. The link type is `Pipe`. It seems that WNTR uses the `Valve` type for special valves like PRVs.